packages/block-library/src/navigation/index.php

```diff
@@ -214,8 +216,17 @@ function( $block ) {
 		$inner_blocks = new WP_Block_List( $compacted_blocks, $attributes );
 	}
 
+	// If there are no inner blocks then fallback to rendering the Page List block.
 	if ( empty( $inner_blocks ) ) {
-		return '';
+		$is_fallback            = true; // indicate we are rendering the fallback.
+		$attributes['maxPages'] = 4; // set value to be passed as context to Page List block.
```

Comment:
Can we drop maxPages as an attribute from this PR and just use the number 4 directly? I'd rather discuss that separately, like what kind of UI is needed to manage that in the editor.

Reply:
I don't think we can easily do that can we? The Page List should only limit the number of items shown when:

- it is inside a Navigation block
- the Nav block has detected no menu is set. 

If we hardcode 4 into the Page List then it will always render only 4 items. That's why I decided to use context.

Open to suggestions.

Reply:
We could add an attribute directly to Page List itself. The block seems to be different though in that it gets all it's properties from context and has no attributes of its own. 

This seems like an intentional design decision so I'd be loath to override that without speaking to someone who worked on that. Perhaps @tellthemachines?

Reply:
Ah you are right, only the attributes are passed down via a context. Hmmm... I guess there isn't much choice then, is there?

Reply:
> I guess there isn't much choice then, is there?

That's where I got to. It's either use context (as I'm doing here) or we add an attribute directly to the `core/page-list` block and set it from the Nav block.

Reply:
I wonder if the Query Loop block would provide maxPages "for free". Maybe it could even remove the need for a separate "page-list" block? Let's move on with an unstable attribute and follow up with query loop explorations.

Reply:
If we really don't like `maxPages` let's:

- revert `maxPages`.
- merge this PR
- raise a follow up _with_ `maxPages` for consideration.

Reply:
`__unstable` does the trick and makes it easy to change the implementation if needed. `maxPages` has to be at least a `page-list` attribute, there is no getting rid of it easily.